src/containers/BooksList.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Book from '../components/Book';
+
+const BooksList = props => {
+  const { books } = props;
+  return (
+    <table>
+      <thead>
+        <tr>
+          <th>Book ID</th>
+          <th>Title</th>
+          <th>Category</th>
+        </tr>
+      </thead>
+      <tbody>
+
+        {
+          books.map(
+            book => <Book key={book.id} book={book} />,
+          )
+        }
+
+      </tbody>
+    </table>
+  );
+};
+
+BooksList.propTypes = {
+  books: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      book: PropTypes.shape({
+        id: PropTypes.number.isRequired,
```

Comment:
- [ ] I don't see why you have nested `shape` proptypes or multiple `id`'s?. According to the shape of books object defined in `App.js`, our Booklist proptypes should be:

```js
BooksList.propTypes = {
  books: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      category: PropTypes.string.isRequired,
    }),
  ).isRequired,
};
```
This raised an error console "Failed Prop Type"
![Screenshot from 2021-01-07 22-38-48](https://user-images.githubusercontent.com/4246637/103942495-26bf7780-5139-11eb-9c5d-2150a9074dea.png)

Kindly fix the console error.

Reply:
Yep @mosaaleb, completely right. That mistake must have crept in at some point while I was developing it. The propTypes statement now reads as follows and there is no console error.

```
BooksList.propTypes = {
  books: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      category: PropTypes.string.isRequired,
    }).isRequired,
  ).isRequired,
};
```